riskive/int-azure-azure-sentinel
# Fix Connector Timeout - Internal PR - Public Repo

- [X] unit test passed
- [X] less than x (800 lines?)

## Purpose

To cover timeouts present in several AzureFunction-based connectors and include async logic to improve performance

## Approach

- Modified SentinelConnector class so it sends batches asynchronously
- Changed ZeroFox connector so it generates batches of result pages instead of entries
- adapted all connectors accordingly

## Where should the reviewer start?
Start with SentinelConnector class, move on to ZeroFoxClient class and then to any connector (they are essentially the same logic against different endpoints)


## This PR addresses a:

- [ ] Vulnerability


## Quality assurance (mandatory)

Created a local azure instance and configured connector to validate all logs appeared

## Steps to validate/test  
  
1.  Set up azure instance
2. make a custom deploy using `mainTemplate.json` file
3.   make a deploy to azure action, with code pointed to zip file
4. configure connector with credentials and test each connector
5. after a few minutes, it should appear as all having received data

### current behavior
### new behavior
